test/parallel_api/numeric/numeric.ops/in_place_scan.pass.cpp

```diff
@@ -19,64 +19,570 @@
+        exec.queue().wait_and_throw();
+
+        retrieve_data(host_keys, host_vals);
+        check_values(host_keys.get(), host_vals.get(), n, BinaryOperation());
+    }
+#endif
+
+    // specialization for host execution policies
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Size>
+    typename ::std::enable_if<
+#if TEST_DPCPP_BACKEND_PRESENT
+        !oneapi::dpl::__internal::__is_hetero_execution_policy<typename ::std::decay<Policy>::type>::value &&
 #endif
+            is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator1>::value,
+        void>::type
+    operator()(Policy&& exec,
+               Iterator1 keys_first, Iterator1 keys_last,
+               Iterator2 vals_first, Iterator2 vals_last,
+               Size n)
+    {
+        typedef typename ::std::iterator_traits<Iterator1>::value_type KeyT;
+
+        // call algorithm with no optional arguments
+        initialize_data(keys_first, vals_first, n);
+        auto res1 = oneapi::dpl::inclusive_scan(exec, keys_first, keys_last, vals_first);
+        check_values(keys_first, vals_first, n);
+
+        // call algorithm with equality comparator
+        initialize_data(keys_first, vals_first, n);
+        auto res2 = oneapi::dpl::inclusive_scan(exec, keys_first, keys_last, vals_first, DefaultBinaryOp<KeyT>());
+        check_values(keys_first, vals_first, n);
+
+        // call algorithm with addition operator
+        initialize_data(keys_first, n);
+        auto res3 = oneapi::dpl::inclusive_scan(exec, keys_first, keys_last, vals_first, BinaryOperation());
+        check_values(keys_first, vals_first, n, BinaryOperation());
+    }
 
-int main()
+    // specialization for non-random_access iterators
+    template <typename Policy, typename Iterator1, typename Iterator2, typename Size>
+    typename ::std::enable_if<!is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator1>::value, void>::type
+    operator()(Policy&& exec,
+               Iterator1 keys_first, Iterator1 keys_last,
+               Iterator2 vals_first, Iterator2 vals_last,
+               Size n)
+    {
+    }
+};
+
+DEFINE_TEST_1(test_inclusive_scan_inplace, BinaryOperation)
 {
+    DEFINE_TEST_CONSTRUCTOR(test_inclusive_scan_inplace)
+
+    // TODO: replace data generation with random data and update check to compare result to
+    // the result of a serial implementation of the algorithm
+    template <typename Iterator1, typename Size>
+    void
+    initialize_data(Iterator1 host_keys, Size n)
+    {
+        for (Size i = 0; i != n; ++i)
+            host_keys[i] = i;
+    }
+
+    // Required to pass source_host_keys_state param by copy for safe state of this param on caller side
+    template <typename ValT, typename Iterator1, typename Size,
+              typename BinaryOperationCheck = oneapi::dpl::__internal::__pstl_plus>
+    void check_values(std::vector<ValT> source_host_keys_state, Iterator1 host_keys, Size n,
```

Comment:
it make sense to pass `std::vector<ValT>` by reference, not by value.

Reply:
Fixed.